### PR TITLE
[INFRA-266] Adjust test for HTTPS core download URLs

### PR DESCRIPTION
**Pull request by timja**
See [INFRA-266](https://issues.jenkins-ci.org/browse/INFRA-266).

Update center now serves plugins over https and so do mirrors.

Possibly the test should just be deleted? Any thoughts?

### Proposed changelog entries

* Entry 1: Issue, Human-readable Text
* ...

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [x] (If applicable) Jira issue is well described
- [x] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change). [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [x] Appropriate autotests or explanation to why this change has no tests
- [x] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [x] There are at least 2 approvals for the pull request and no outstanding requests for change
- [x] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [x] Changelog entries in the PR title and/or `Proposed changelog entries` are correct
- [x] Proper changelog labels are set so that the changelog can be generated automatically
- [x] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [x] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins-ci.org/issues/?filter=12146)). [JENKINS-63484](https://issues.jenkins-ci.org/browse/JENKINS-63484)

### CODE REVIEW

File: test/src/test/java/hudson/model/UpdateCenterTest.java
Comment (timja) [inline test/src/test/java/hudson/model/UpdateCenterTest.java:66]: the issue was very obvious after changing to assertThat rather than assert true

```
java.lang.AssertionError: 
Expected: a string starting with "http://updates.jenkins-ci.org/"
     but: was "https://updates.jenkins.io/download/war/2.253/jenkins.war"
```

**Review by daniel-beck — APPROVED** [review]

**Comment by daniel-beck** [comment]
CC @olivergondza this will also need to be in LTS.

**Review by res0nance — APPROVED** [review]

**Review by MarkEWaite — APPROVED** [review]

**Review by oleg-nenashev — APPROVED** [review]
We may merge it in 24 hours if there is no negative feedback. Please see the [merge process documentation](https://github.com/jenkinsci/jenkins/blob/master/docs/MAINTAINERS.adoc#merge-process) for more information about the merge process

**Comment by MarkEWaite** [comment]
I propose we accept this merge as an intentional exception to the 24 hours before merge process so that we can merge it as soon as the CI process is successful.

The change is specifically a test and it is a test fix based on improved infrastructure (https instead of http) that we have wanted for a very long time.

**Comment by jglick** [comment]
> Possibly the test should just be deleted?

Yes, it is mainly a test of an external service and so should be moved elsewhere. Anyway, let us get this merged for now.

**Review by jglick — APPROVED** [review]

**Pull request merged**
